pkg/apis/mattermost/v1alpha1/clusterinstallation_utils.go

```diff
@@ -514,23 +521,41 @@ func (mattermost *ClusterInstallation) GenerateDeployment(deploymentName, ingres
 		Value: valueSize,
 	})
 
-	// Mattermost License
-	volumeLicense := []corev1.Volume{}
-	volumeMountLicense := []corev1.VolumeMount{}
+	// Mattermost container volumes
+	volumes := []corev1.Volume{}
+	volumesMount := []corev1.VolumeMount{}
 	podAnnotations := map[string]string{}
+
+	// CA Certificate Files
+	volumesMount = append(volumesMount, corev1.VolumeMount{
+		Name:      CertificateFilesSecretName,
+		MountPath: CertificateFilesPath,
+		ReadOnly:  true,
+	})
+
+	volumes = append(volumes, corev1.Volume{
+		Name: CertificateFilesSecretName,
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName: CertificateFilesSecretName,
```

Comment:
if the secret does not exist this will fail and the pod will not start. We need to validate if the secret exist and have the correct format

Reply:
Also if you mount this volume in the `/etc/ssl/certs` if going to overwrite all the other certificates we install when we build the Mattermost image, [see](https://github.com/mattermost/mattermost-server/blob/master/build/Dockerfile#L12)

to avoid this you will need to mount like this
```
    volumeMounts:
    - name: foo
      mountPath: /etc/ssl/certs/rds-ca-2019-root.pem
      subPath: rds-ca-2019-root.pem
      readOnly: true
```

Reply:
Right, that's what I had before. I do the change!

Reply:
the `CertificateFilesSecretName` should be configurable as well and if not set it can use the default value

Reply:
I was thinking about using this constant in `mattermost-cloud` but if we are going to make this configurable, it changes everything.